### DISCUSSION TITLE
Listing inventory returned with duplicate products (variations)

### DISCUSSION THREAD

**Discussion by nmrdxt** [etsy/open-api#667] [Q&A]
Hello,

I'm seeing some Etsy listings returned with duplicate products. 

**To replicate:**

Fetch this listing's inventory: 878027322 (ID)

Listing should have 77 products, but 308 are returned. With each product (property values combo) duplicated 4 times. Each product has its unique ID (and unique offering ID), and all are enabled (and not deleted). It looks like a bug, but I might be missing something.

I'd appreciate assistance with this – thanks!

**Comment by nmrdxt** [comment]
@etsyachristensen I'd love to get an input regarding this, when you get a chance, as I'm seeing this with more products, and it might affect others as well. Thanks!

**Reply by etsyachristensen** [reply]
I just queried that listing's inventory and I get 22 products returned.

**Reply by nmrdxt** [reply]
Thanks for checking @etsyachristensen, and sorry for the delayed response!

The merchant has updated the listing's inventory in that 8 days gap between my original request and the time you have tested it.

I'm seeing this with another product.

Listing ID is 1361553423

Its inventory has multiple products with the property value combo of Tulsi Teas / 10 Plain Bags. Each product with a different ID, but same price and quantity.

This seems like a bug, if it can be checked and hopefully resolved it would be awesome – thanks!

**Reply by etsyachristensen** [reply]
Ah yes. I see it in this listing now. I see the last 3 products are the Tulsi / 10 Bag combo. We'll see what we can track down.

**Reply by nmrdxt** [reply]
Thanks! Much appreciated. There are more examples, if needed.

**Reply by etsyachristensen** [reply]
Just wanted to note that in our logs, the only API activity for that listing is going through V2, not V3. It doesn't appear that inventory has been updated via the API yet though so I'm wondering if updating it via V3 might fix the problem by removing the extra entries. Obviously this isn't a solution, but a temporary measure your app can take.

**Reply by nmrdxt** [reply]
Thank you for the suggestion – our App has yet to update the inventory, since it is unable to determine which product is the correct one (due to the duplication).

I'll try to send an inventory update request, using the same object received from Etsy, hopefully it will resolve the issue.

But, I agree – it's a temporary measure, it would be great if this can be resolved.

Thanks!

**Reply by etsyachristensen** [reply]
It shouldn't matter which one of those you keep.

**Reply by nmrdxt** [reply]
Hey, Are there any updates regarding this? I've tried updating the inventory in various ways, but the duplicate products remain. 

Is there a plan to remove all the duplicate products? Because if there is a plan to remove duplicate products, it matters which one of the products we keep.

Thanks!

**Reply by etsyachristensen** [reply]
At this time, there's no automated mechanism for us to remove these duplicates. It could be that some of these entries were supposed to have been deleted in a previous inventory update. It's possible that V2 inventory system isn't removing all the variations when you update your inventory there. From our perspective it doesn't matter which ones you remove when you send in the updated inventory data record via V3 because the system replaces the existing inventory data with the previous data. You'll find that new offering ids and product ids are generated after updating inventory. Our system works on the property id and their values to determine uniqueness.

**Comment by uzairfiaz-gelato** [comment]
Hi @nmrdxt , @nateswart 
Any update on this? We are also getting the same issue. For listing Id: 1291864218

Thanks
